File: abc119_c.py
"""
C - Synthetic Kadomatsu
https://atcoder.jp/contests/abc119/tasks/abc119_c
あなたは N 本の竹を持っています。これらの長さはそれぞれ l1,l2,...,lN です (単位: センチメートル)。
あなたの目的は、これらの竹のうち何本か (全部でもよい) を使い、長さが A,B,C であるような 3 本の竹を得ることです。
そのために、以下の三種類の魔法を任意の順に何度でも使うことができます。
- 延長魔法: 1 MP (マジックポイント) を消費し、1 本の竹を選んでその長さを 1 増やす。
- 短縮魔法: 1 MP を消費し、1 本の長さ 2 以上の竹を選んでその長さを 1 減らす。
- 合成魔法: 10 MP を消費し、2 本の竹を選んで接続し 1 本の竹とする。この新たな竹の長さは接続した 2 本の竹の長さの合計に等しい。(以後、この竹に対してさらに魔法を使用することもできる。)
目的を達成するには、最小でいくつの MP が必要でしょうか？
"""
"""
N, A, B, C = map(int, input().split())
l = [int(input()) for i in range(N)]
INF = 10 ** 9
def dfs(cur, a, b, c):
    if cur == N:
        return abs(a - A) + abs(b - B) + abs(c - C) - 30 if min(a, b, c) > 0 else INF
    ret0 = dfs(cur + 1, a, b, c)
    ret1 = dfs(cur + 1, a + l[cur], b, c) + 10
    ret2 = dfs(cur + 1, a, b + l[cur], c) + 10
    ret3 = dfs(cur + 1, a, b, c + l[cur]) + 10
    return min(ret0, ret1, ret2, ret3)
print(dfs(0, 0, 0, 0))
"""
N, A, B, C = map(int, input().split())
l = [int(input()) for _ in range(N)]
def dfs(cur, a, b, c):
    if cur == N:
        return abs(a-A) + abs(b-B) + abs(c-C) - 30 if a * b * c != 0 else 10 ** 9
    else:
        ret0 = dfs(cur + 1, a, b, c)
        reta = dfs(cur + 1, a + l[cur], b, c) + 10
        retb = dfs(cur + 1, a, b + l[cur], c) + 10
        retc = dfs(cur + 1, a, b, c + l[cur]) + 10
        return min(ret0, reta, retb, retc)
print(dfs(0,0,0,0))

File: abc119_c.cpp
// https://atcoder.jp/contests/abc119/tasks/abc119_c
#include<iostream>
// #include<algorithm>
// #include<string>
// #include<numeric>
// #include<vector>
// #include<map>
// #include<tuple>
// #include<set>
// #include<queue>
// #include<deque>
// #include<regex>
// #include<bitset>
// #include<iomanip>
// #include<complex>
// #include<stack>
// #include<functional>

#include<bits/stdc++.h>
using namespace std;
#define rep(i,n) for (int i = 0; i < (n); ++i)
#define all(v) (v).begin(),(v).end()
typedef pair<int, int> P;
typedef long long ll;
const int INF = 1001001001;
const ll LINF = 1001002003004005006ll;
const ll MOD = 1e9+7;

int n;
vector<int> abc(3); // A, B, C を配列で持つことで、ループ処理を可能にする
int ans = INF;
vector<P> now(4, P(0, 0));  // 長さの合計、使用本数をpairで持つ

void dfs(int depth, vector<int> &l) {
    if (depth == n) {
        int tmp = 0;
        rep(i, 3) {
            if (now[i].second == 0) return;
            // 目的の長さとの差 + 合成魔法で使ったMP
            tmp += abs(abc[i] - now[i].first) + 10 * (now[i].second - 1);
        }
        ans = min(ans, tmp);
        return;
    }
    rep(i, 4) { // 各竹は4つのグループ(A,B,C,未使用)に入る可能性がある
        now[i].first += l[depth];
        now[i].second += 1;
        dfs(depth + 1, l);
        now[i].first -= l[depth];   // 抜けたら選択を戻す
        now[i].second -= 1;
    }
}

int main() {
	cin >> n;
    rep(i, 3) cin >> abc[i];
    vector<int> l(n);
    rep(i, n) cin >> l[i];
    dfs(0, l);
	cout << ans << endl;
	return 0;
}
